### DISCUSSION TITLE
Selecting lowercase characters also matches with the upper case but not the other way around

### DISCUSSION THREAD

**Discussion by fklose** [helix-editor/helix#11982] [General] [answered]
### Summary

I tried this:

1. Open an empty buffer.
2. Write `aA` in insert mode.
3. In normal mode highlight the line using `x` and select for `a` using `s` to enter selection -> `a` -> Enter

I expected this to only match the lowercase letter but it also matches the uppercase. When I tried selecting `A` it does not match the lower case letter.

*Note:* I am not familiar with what to expect from regex, when I try the same thing in an online regex interpreter it matches the cases properly, meaning that selecting `a` only highlights `a` and the same for `A`.

### Platform

Linux

### Terminal Emulator

zsh 5.9 (x86_64-debian-linux-gnu)

### Installation Method

releases page

### Helix Version

helix 24.7 (079f5442)

**Comment by pascalkuthe** [comment, marked as answer]
That's called smart-case and intentional. It will be case insensitive if everything is lower case and case sensitive if there is any uppercase character in the regex. You can prefix your regex with `(?-i)` to disable that behavior or disable the corresponding config option

**Reply by fklose** [reply]
I was not aware of that. Thank you for your quick reply.
